test/api/shorten.test.js

```diff
@@ -0,0 +1,226 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+import supertest from 'supertest';
+import nock from 'nock';
+
+import { config } from '../../src/config';
+import { createApp } from '../../src/app';
+import { ACCEPT_VALUE_MIME } from '../../src/middlewares/versioning';
+import {
+  checkSecurityHeaders,
+  checkCorsHeader,
+} from './utils/check-security-headers';
+
+const BITLY_HOSTNAME = 'https://api-ssl.bitly.com';
+
+describe('shorten url', () => {
+  function getPreconfiguredRequest() {
+    const acceptHeader = ACCEPT_VALUE_MIME + ';version=1';
+    const agent = supertest(createApp().callback());
+    return agent
+      .post('/shorten')
+      .accept(acceptHeader)
+      .type('json');
+  }
+
+  function setup({ longUrl, shortUrl } = {}) {
+    const nockScope = nock(BITLY_HOSTNAME, {
+      reqheaders: { authorization: `Bearer ${config.bitlyToken}` },
+    })
+      .post('/v4/shorten', { long_url: longUrl })
+      .reply(201, { link: shortUrl });
+
+    const request = getPreconfiguredRequest();
+    return { request, nockScope };
+  }
+
+  it('calls the underlying service to shorten an url', async () => {
+    const longUrl = 'https://profiler.firefox.com/public/FAKEHASH/calltree';
+    const shortUrl = 'https://share.firefox.dev/BITLYHASH';
+
+    const { request, nockScope } = setup({ longUrl, shortUrl });
+
+    await request.send({ longUrl }).expect(200, { shortUrl });
+    nockScope.done();
+  });
+
+  it('rejects with an error if the request is badly formed', async () => {
+    // Silence the error reporter.
+    jest.spyOn(process.stdout, 'write').mockImplementation(() => {});
+
+    const { request, nockScope } = setup();
+
+    // Not a JSON
+    await request.send('a').expect(400);
+
+    // An error has been output to the console.
+    expect(process.stdout.write).toHaveBeenCalledWith(
+      expect.stringContaining('server_error')
+    );
+
+    // There's been no request to the underlying service.
+    expect(nockScope.isDone()).toBe(false);
+  });
+
+  it('rejects with an error if the request does not have the expected properties', async () => {
```

Comment:
Optional: If implementing a Bitly API error class like I suggested, then, test for the error response:

```
{
  "message": "string",
  "errors": [
    {
      "field": "string",
      "message": "string",
      "error_code": "string"
    }
  ],
  "resource": "string",
  "description": "string"
}
```